interfaces/apparmor/template_vars.go

```diff
@@ -88,22 +97,23 @@ func modernVariables(appInfo *snap.AppInfo) string {
 // profile. That same content also decides if the profile is enforcing or
 // advisory (complain). This is used to implement developer mode.
 func (b *Backend) aaHeader(appInfo *snap.AppInfo, developerMode bool) []byte {
-	var text string
-	// Use a different template when requested.
-	if len(b.CustomTemplate) == 0 {
-		text = strings.TrimRight(defaultTemplate, "\n}")
-		// TODO: switch to modernVariables() after the default template is
-		// compatible with them.
-		text = strings.Replace(text, "###VAR###\n", legacyVariables(appInfo), 1)
-	} else {
-		text = strings.TrimRight(b.CustomTemplate, "\n}")
-		text = strings.Replace(text, "###VAR###\n", legacyVariables(appInfo), 1)
+	template := b.legacyTemplate
+	if template == nil {
+		template = defaultTemplate
 	}
+	template = bytes.TrimRight(template, "\n}")
```

Comment:
This is sadly required. There's no other ### thing that we can use reliably to insert our snippets. This is the same code that was done earlier to insert apparmor rules for hw-assign. Unless we are willing to break 15.04 compatibility and require a new block for snippets something like this has to stay around.